src/nrniv/nrndae.cpp

```diff
@@ -30,9 +31,12 @@ int nrndae_extra_eqn_count() {
 }
 
 void nrndae_update() {
+    NrnThread* _nt = nrn_threads;
+    update_sp13_rhs_based_on_actual_rhs(_nt);
```

Comment:
Should this be a loop over threads?

Reply:
It's not needed because there is already a high level loop that goes through all the `NrnThread`s. See https://github.com/neuronsimulator/nrn/blob/3360cd6531d3f7502c59f4d0e8ee92a0f48abe3b/src/nrnoc/multicore.cpp#L892, https://github.com/neuronsimulator/nrn/blob/a66b72e5e0dcd0626af61311ad3574186d35c0b6/src/nrnoc/fadvance.cpp#L318, https://github.com/neuronsimulator/nrn/blob/a66b72e5e0dcd0626af61311ad3574186d35c0b6/src/nrnoc/fadvance.cpp#L299, https://github.com/neuronsimulator/nrn/blob/a66b72e5e0dcd0626af61311ad3574186d35c0b6/src/nrnoc/solve.cpp#L343

Reply:
OK, but don't we need to pass in the `NrnThread` from those higher level loops?
e.g. for `nrndae_update` we need to make it take `NrnThread&` and pass in the `nt` argument in `void nrn_update_voltage(neuron::model_sorted_token const& sorted_token, NrnThread& nt)` where `nrndae_update` is called? `nrn_threads` is always the zeroth thread.

Reply:
Now I see what you mean, yes you're right :+1: